### PR TITLE
feat: Allow for ad-related props to be changed

### PR DESCRIPTION
**Pull request by fabiosantoscode**
*    BREAKING CHANGE: "animated" and "lazyLoad" props have been deprecated
 *    BREAKING CHANGE: sizes and sizeMappings defaults have been removed

This allows for an AdPanel's adTag to be changed during runtime. Previously, changing the adTag had no effect.

This also decimates some old stuff which was not working anyways.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -152,7 +120,6 @@ describe('AdPanel', () => {
           [ 'baz', 'qux' ],
         ];
         googleTag = createFakeGoogleTag();
-        AdPanel.getOrCreateGoogleTag = chai.spy(() => googleTag);
```
Comment (fabiosantoscode) [inline test/index.js]: Removed it, because it had already been defined in the top-level `describe`

File: src/index.js [hunk 2]
```diff
@@ -125,54 +120,48 @@ export default class AdPanel extends React.Component {
...
+      };
+      /* eslint-enable arrow-body-style */
+      const didAnyAdRelatedPropChange =
+        didChange('adTag') ||
+        didChange('targeting') ||
+        didChange('sizes') ||
+        didChange('sizeMapping');
+      if (didAnyAdRelatedPropChange) {
+        const googleTag = AdPanel.getOrCreateGoogleTag();
+        this.destroySlot();
```
Comment (gcedo) [inline src/index.js]: This is awesome!

**Comment by gcedo** [comment]
I love decimation, but I think this component is shared with the WorldIf, which uses the lazyload features.

**Comment by fabiosantoscode** [comment]
I hate to break backwards compatibility, but I feel there's lots to gain in this case. I've talked to @angelorepole about this, and he asked that I document the new way of using LazyLoad, which I just did in the latest commit.

**Comment by gcedo** [comment]
Ok, I'm ready to merge, can @umbobabo acknowledge?

**Pull request merged**